src/common/System.cpp

```diff
@@ -224,6 +226,8 @@ void Drop(Str::StringRef message)
 {
 	if (!OnMainThread()) {
 		Sys::Error(message);
+	} else {
+		PrintStackTrace();
```

Comment:
I got stack overflows in some tests due to IPC errors while trying to print the stack trace leading to recursion. So this should go after the recursive error test below